Tasks/AzurePowerShellV3/TryMakingModuleAvailable.ps1

```diff
@@ -0,0 +1,33 @@
+[CmdletBinding()]
+param (
+    [string]
+    $targetVersion
+)
+
+. "$PSScriptRoot\Utility.ps1"
+$moduleContainerPath = Get-SavedModuleContainerPath
+
+if (-not(Test-Path -Path $moduleContainerPath)) {
+    return;
+}
+
+if (!$targetVersion) {
+    return;
+}
+
+# value for classic
+@($false, $true).ForEach({
+    $modulePath = Get-SavedModulePath -azurePowerShellVersion $targetVersion -Classic:$_;
```

Comment:
Can we check whether classic or ARM connection is used and then unzip appropriate module instead of unzipping both the azure and azurerm modules? you can find this logic here https://github.com/microsoft/azure-pipelines-tasks/blob/693e719b4dbf83dc3637a040273035de9bb5221e/Tasks/Common/VstsAzureHelpers_/VstsAzureHelpers_.psm1#L27

Reply:
I checked, ideally we might be able to but that seems out of scope of the PR since we the connection check is done in `Initialize-Azure` which is called after the call to `Update-PSModulePathForHostedAgent` but we need to make the module available before the call to `Update-PSModulePathForHostedAgent`. Making these changes doesn't seem ideal or needed as of now as it involves changing of existing logic.